lib/src/solver/solve_suggestions.dart

```diff
@@ -0,0 +1,214 @@
+            } else {
+              priority = 2;
+              suggestion =
+                  '* Try upgrading your constraint on $name: $topLevelProgram pub add $addDescription';
+            }
+          }
+        }
+
+        return _ResolutionSuggestion(suggestion, priority: priority);
+      }
+    }
+    return null;
+  }
+
+  Future<_ResolutionSuggestion?> suggestUnlockingAll(
+      {required bool stripLowerBound}) async {
+    final originalPubspec = entrypoint.root.pubspec;
+    final relaxedPubspec =
+        stripVersionBounds(originalPubspec, stripLowerBound: stripLowerBound);
+
+    final result = await tryResolve(
+        type, cache, Package.inMemory(relaxedPubspec),
+        lockFile: entrypoint.lockFile, unlock: unlock);
+    if (result != null) {
+      final updatedPackageVersions = <PackageId>[];
+      for (final id in result.packages) {
+        final originalConstraint = (originalPubspec.dependencies[id.name] ??
+                originalPubspec.devDependencies[id.name])
+            ?.constraint;
+        if (originalConstraint != null) {
+          updatedPackageVersions.add(id);
+        }
+      }
+      if (stripLowerBound && updatedPackageVersions.length > 5) {
+        // Too complex, don't suggest.
+        return null;
+      }
+      if (stripLowerBound) {
+        updatedPackageVersions.sort((a, b) => a.name.compareTo(b.name));
+        final formattedConstraints = updatedPackageVersions
+            .map((e) => packageAddDescription(entrypoint, e))
+            .join(' ');
+        return _ResolutionSuggestion(
+            '* Try updating the following constraints: $topLevelProgram pub add $formattedConstraints',
+            priority: 4);
+      } else {
+        return _ResolutionSuggestion(
+            '* Try an upgrade of your constraints: $topLevelProgram pub upgrade --major-versions',
+            priority: 4);
+      }
+    }
+    return null;
+  }
+
+  final visited = <String>{};
+  final stopwatch = Stopwatch()..start();
+  final suggestions = <_ResolutionSuggestion>[];
+  void addSuggestion(_ResolutionSuggestion? suggestion) {
+    if (suggestion != null) suggestions.add(suggestion);
+  }
+
+  for (final externalIncompatibility
+      in incompatibility.externalIncompatibilities) {
+    if (stopwatch.elapsed > Duration(seconds: 3)) {
+      // Never spend more than 3 seconds computing suggestions.
+      break;
+    }
+    final cause = externalIncompatibility.cause;
+    if (cause is SdkCause) {
+      addSuggestion(await suggestSdkUpdate(cause));
+    } else {
+      for (final term in externalIncompatibility.terms) {
+        final name = term.package.name;
+
+        if (!visited.add(name)) {
+          continue;
+        }
+        addSuggestion(await suggestSinglePackageUpdate(name));
+      }
+    }
+  }
+  if (suggestions.isEmpty) {
+    addSuggestion(await suggestUnlockingAll(stripLowerBound: true) ??
+        await suggestUnlockingAll(stripLowerBound: false));
```

Comment:
Shouldn't this be the other way around?

Reply:
We first attempt stripping only the upper bound, if that fails we relax further and strip the lower bound as well...